### ISSUE TITLE
Should accept and use jaguar_client under the hood

### ISSUE THREAD

**Issue by winterDroid** [Jaguar-dart/jaguar_http#4]
Since jaguar_client provides already features like cookie handling, I was wondering whether it is planned to accept it as well as normal client?

**Comment by tejainece** [comment]
> whether it is planned to accept it as well as normal client?

I am sorry. Don't understand you here. Can you please rephrase?

You are welcome to chat with us on gitter:

https://gitter.im/jaguar_dart/jaguar

**Comment by tejainece** [comment]
Cool idea! We have to choose one or the other tho. 

Either use http::Client or jaguar_client::JsonClient.

**Comment by lejard-h** [comment]
can't we have `jaguar_client::JsonClient` extending `http::Client` ?
limiting to `JsonClient` mean that we can only manipulate JSON data.

**Comment by tejainece** [comment]
Namespace collision. Both of them have `get`, `post`, etc with different signatures.

https://github.com/dart-lang/http/blob/master/lib/src/base_client.dart#L21

https://github.com/Jaguar-dart/jaguar_client/blob/master/lib/src/client.dart#L52

Could have named them `getJson`, `postJson`, etc in `jaguar_client::JsonClient` but short and straight forward names made more sense.

Alternatively, you can write two classes in `jaguar_http`. Depending on the supplied client, the factory constructor returns correct sub-class.

**Comment by lejard-h** [comment]
```
class JaguarClient {}
class JsonClient extends JaguarClient {}

final client = new JaguarClient(new BrowserClient());
final clientJson = new JsonClient(new BrowserClient());
```

Something like that could be the solution, we want to manipulate only json

**Comment by lejard-h** [comment]
But we could also merge both library, `jaguar_http` is already using `jaguar_serializer` and `jaguar_client` provide an abstraction hover `http` package.

**Comment by tejainece** [comment]
I agree. Lets merge the annotation part into `jaguar_client` and use this repo for the CLI/generator.
